hydra-node/src/Hydra/HeadLogic.hs

```diff
@@ -236,13 +236,19 @@ deriving instance (ToJSON (Event tx), ToJSON (HeadState tx)) => ToJSON (LogicErr
 deriving instance (FromJSON (Event tx), FromJSON (HeadState tx)) => FromJSON (LogicError tx)
 
 data Outcome tx
-  = OnlyEffects [Effect tx]
-  | NewState (HeadState tx) [Effect tx]
-  | Wait WaitReason
-  | Error (LogicError tx)
+  = OnlyEffects {effects :: [Effect tx]}
+  | NewState {headState :: HeadState tx, effects :: [Effect tx]}
+  | Wait {reason :: WaitReason}
+  | Error {error :: LogicError tx}
+  deriving stock (Generic)
 
 deriving instance (IsTx tx, IsChainState tx) => Eq (Outcome tx)
 deriving instance (IsTx tx, IsChainState tx) => Show (Outcome tx)
+deriving instance (IsTx tx, IsChainState tx) => ToJSON (Outcome tx)
+deriving instance (IsTx tx, IsChainState tx) => FromJSON (Outcome tx)
+
+instance (IsTx tx, Arbitrary (ChainStateType tx)) => Arbitrary (Outcome tx) where
```

Comment:
Could: Use `IsChainState tx` here as well instead of `Arbitrary (ChainStateType tx)`

Reply:
cant do it 🤷 

this is also how it is defined the instance for: 
- PostTxError
- HeadState
- HydraNodeLog

tried to change them as well and did not work either.

how would you perform it?

Reply:
Add a `Arbitrary (ChainStateType tx)` to the super class constraints of `IsChainState` :trollface: I just realized its not their yet.. fine to keep like it is.